### PR TITLE
add to MSAL contributing guide

### PR DESCRIPTION
**Pull request by kellyyangsong**
Adapted the team contributing guide to the one in MSAL. Did not remove anything from the original MSAL guide as it had a lot of great information, and I think the newest additions plug in nicely.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,6 +31,70 @@ git checkout joes_feature_branch
...
+
+Very large PRs that would take hours to review (remember, we're trying to help lots of people at once). For larger work areas, please discuss with us to find ways of breaking it down into smaller, incremental pieces that can go into separate PRs.
+
+## During pull request review
+
+A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
+
+## Submitting a pull request
+
+If you're not sure how to create a pull request, read this article: https://help.github.com/articles/using-pull-requests. Make sure the repository can build and all tests pass. Familiarize yourself with the project workflow and our coding conventions. The coding, style, and general engineering guidelines are published on the Engineering guidelines page.
```
Comment (pmaytak) [inline CONTRIBUTING.md:79]:
```suggestion
If you're not sure how to create a pull request, read GitHub's [About pull requests
](https://help.github.com/articles/using-pull-requests) article. Make sure the repository can build and all tests pass. Familiarize yourself with the project workflow and our coding conventions. The coding style and general engineering guidelines are published on the Engineering guidelines page.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -31,6 +31,70 @@ git checkout joes_feature_branch
...
+
+## Writing Tests
+
+- Tests need to be provided for every bug/feature that is completed.
+- Tests need to be provided for every bug/feature that is completed.
+  - Unit tests cover all new aspects of the code.
+- Before and after performance and stress tests results are evaluated (no regressions allowed).
```
Comment (pmaytak) [inline CONTRIBUTING.md:86]:
```suggestion
- Tests need to be provided for every bug and feature that is completed.
  - Unit tests must cover all new aspects of the code.
- Before and after performance and stress tests results are successfully evaluated - no regressions are allowed.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -31,6 +31,70 @@ git checkout joes_feature_branch
...
+
+## During pull request review
+
+A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
+
+## Submitting a pull request
+
+If you're not sure how to create a pull request, read this article: https://help.github.com/articles/using-pull-requests. Make sure the repository can build and all tests pass. Familiarize yourself with the project workflow and our coding conventions. The coding, style, and general engineering guidelines are published on the Engineering guidelines page.
+
+## Writing Tests
```
Comment (pmaytak) [inline CONTRIBUTING.md:81]:
```suggestion
## Ensuring test coverage
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -31,6 +31,70 @@ git checkout joes_feature_branch
...
+
+- Tests need to be provided for every bug/feature that is completed.
+- Tests need to be provided for every bug/feature that is completed.
+  - Unit tests cover all new aspects of the code.
+- Before and after performance and stress tests results are evaluated (no regressions allowed).
+- Performance and stress tests are extended as relevant.
+
+## Feedback
```
Comment (pmaytak) [inline CONTRIBUTING.md:89]:
```suggestion
## Pull request feedback
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -31,6 +31,70 @@ git checkout joes_feature_branch
...
+
+Your pull request will now go through extensive checks by the subject matter experts on our team. Please be patient; we have hundreds of pull requests across all of our repositories. Update your pull request according to feedback until it is approved by one of the team members.
+
+## Merging pull requests
```
Comment (pmaytak) [inline CONTRIBUTING.md:93]:
```suggestion
## Merging a pull request
```

**Review by pmaytak — COMMENTED** [review]
Thanks. I've made some minor changes, including combing the tests sections and moving fork related section after merging a PR section.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -6,13 +6,75 @@ Microsoft Authentication Library (MSAL) for .NET welcomes new contributors.  Thi
...
+- Changes that break backward compatibility
+- Changes that are only wanted by one person or company. Changes need to benefit a large enough proportion of developers using our auth libraries.
+- Changes that add entirely new feature areas without prior agreement
+- Changes that are mostly about refactoring existing code or code style
+
+Very large PRs that would take hours to review (remember, we're trying to help lots of people at once). For larger work areas, please discuss with us to find ways of breaking it down into smaller, incremental pieces that can go into separate PRs.
+
+### During a pull request review
+
+A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
```
Comment (pmaytak) [inline CONTRIBUTING.md:50]: Has this automation been completed in other repos?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -6,13 +6,75 @@ Microsoft Authentication Library (MSAL) for .NET welcomes new contributors.  Thi
...
+
+Note that you won't be able to run integration tests locally because they connect to Azure Key Vault to fetch some test users and passwords.
+
+### Pull request feedback
+
+Your pull request will now go through extensive checks by the subject matter experts on our team. Please be patient; we have hundreds of pull requests across all of our repositories. Update your pull request according to feedback until it is approved by one of the team members.
+
+### Merging a pull request
+
+When your pull request has had all feedback addressed, it has been signed off by one or more reviewers with commit access, and all checks are green, we will commit it.
```
Comment (pmaytak) [inline CONTRIBUTING.md:72]: This is a bit inaccurate since we don't run CI on forks. Idk if we need to rephrase.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -6,13 +6,75 @@ Microsoft Authentication Library (MSAL) for .NET welcomes new contributors.  Thi
 
 Please visit [https://cla.microsoft.com/](https://cla.microsoft.com/) and sign the Contributor License Agreements.  You only need to do that once. We can not look at your code until you've submitted this request.
 
-## Tests
+## Contributing your own code changes
```
Comment (pmaytak) [inline CONTRIBUTING.md:9]: @localden Thoughts on moving the section on Submitting a bug/feature request above this one?

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -6,13 +6,75 @@ Microsoft Authentication Library (MSAL) for .NET welcomes new contributors.  Thi
...
+
+### Identifying scale
+
+If you would like to contribute to one of our repositories, first identify the scale of what you would like to change. If it is small (grammar/spelling or a bug fix), feel free to start working on a fix. If you are submitting a feature or a substantial code contribution, please discuss it with the team and ensure it follows the product roadmap. You might also read these two blogs posts on contributing code: [Open Source Contribution Etiquette by Miguel de Icaza](http://tirania.org/blog/archive/2010/Dec-31.html) and [Don't "Push" Your Pull Requests by Ilya Grigorik](https://www.igvita.com/2011/12/19/dont-push-your-pull-requests/). All code submissions will be rigorously reviewed and tested further by the team, and only those that meet an extremely high bar for both quality and design/roadmap appropriateness will be merged into the source.
```
Comment (pmaytak) [inline CONTRIBUTING.md:27]: Good articles!

Comment (bgavrilMS) [reply CONTRIBUTING.md:27]: +1 , nice articles

**Review by bgavrilMS — APPROVED** [review]

File: CONTRIBUTING.md (hunk 6 above)
Comment (kellyyangsong) [reply CONTRIBUTING.md:50]: No, this has not and I don't think we plan to.

File: CONTRIBUTING.md (hunk 7 above)
Comment (kellyyangsong) [reply CONTRIBUTING.md:72]: @pmaytak I didn't know that! 😁 How would you like to see it rephrased?

**Review by localden — APPROVED** [review]

**Comment by localden** [comment]
This is great, thank you @kellyyangsong!

File: CONTRIBUTING.md (hunk 6 above)
Comment (pmaytak) [reply CONTRIBUTING.md:50]: I've merged this into the Merging a pull request section.

File: CONTRIBUTING.md (hunk 7 above)
Comment (pmaytak) [reply CONTRIBUTING.md:72]: It actually applies to all of our GitHub repos. I've reworded this section.

**Review by pmaytak — APPROVED** [review]

**Pull request merged**